src/components/MovieForm/types.ts

```diff
@@ -1,3 +1,13 @@
 export interface IMovieFormProps {
   type: 'add' | 'edit' | 'delete';
 }
+
+export interface IMovieFormFields {
+  addAndEdit: {
+    label: string;
+    type: 'text' | 'textarea' | 'select' | 'date';
```

Comment:
Like with components, functions, classes or objects, remember to breakdown interfaces and types into smaller ones.
This helps with separation of concerns and keeping DRY.
e.g. `type MovieFormFieldType = 'text' | 'textarea' | 'select' | 'date';`
`export interface IMovieFormFields {
  addAndEdit: {
    label: string;
    type: MovieFormFiledType;
  (...) }`